### ISSUE TITLE
Different include dirs based on whether the package needs to be build first

### ISSUE THREAD

**Issue by Hedanito** [conan-io/conan#2183]
Don't worry, this is a bug (probably), not a request 😄 

So I made a package for glslang that simply downloads the source from github and builds it:
```
from conans import ConanFile, CMake
import os

class glslangConan(ConanFile):
    name = "glslang"
    version = "17.11.10"
    description = "OpenGL and OpenGL ES shader front end and validator"
    license = "BSD (modified)"

    settings = "os", "compiler", "build_type", "arch"

    generators = "cmake"

    def source(self):
        self.run("git clone https://github.com/KhronosGroup/glslang .")
        self.run("git checkout a36997cb4a3235ed6f73986afae73def59672d93")

    def build(self):
        cmake = CMake(self)
        self.run("cmake . %s -DCORE_BUILD_LIBRARIES_ONLY=ON" % cmake.command_line)
        self.run("cmake --build . --target INSTALL %s" % cmake.build_config)

    def package(self):
        self.cpp_info.includedirs = [os.path.join("include", "glslang", "Public"), os.path.join("include", "glslang", "Include"), os.path.join("include", "SPIRV")]
        self.cpp_info.libdirs = ["lib"]
        self.cpp_info.libs = ["glslang", "HLSL", "OGLCompiler", "OSDependent", "SPIRV", "SPVRemapper"]

        if self.settings.build_type == "Debug":
            for i in range(0, len(self.cpp_info.libs)):
                self.cpp_info.libs[i] = self.cpp_info.libs[i] + "d"
```

All of that is working fine. The weird issue has to do with installation of the package. When the package is installed for the first time (with `conan install --build missing`), the package is build and I get the include directories as described in the package function. However, when I install the package again and it uses the cached build, I get different include directories!

First install `conanbuildinfo.txt`:
```
[includedirs]
C:/Users/rick_/.conan/data/glslang/17.11.10/aurion/stable/package/63da998e3642b50bee33f4449826b2d623661505/include/glslang/Public
C:/Users/rick_/.conan/data/glslang/17.11.10/aurion/stable/package/63da998e3642b50bee33f4449826b2d623661505/include/glslang/Include
C:/Users/rick_/.conan/data/glslang/17.11.10/aurion/stable/package/63da998e3642b50bee33f4449826b2d623661505/include/SPIRV
```

Second (and future) install `conanbuildinfo.txt`:
```
[includedirs]
C:/Users/rick_/.conan/data/glslang/17.11.10/aurion/stable/package/63da998e3642b50bee33f4449826b2d623661505/include
```

I have no idea how this is happening. It took a while to even figure out why I had to delete and rebuild the package every time I installed it. I need help 😭

Some other info:
 * Platform: **Windows 10**
 * Conan version: **0.30.3**

**Comment by Hedanito** [comment]
Of course, the second I click submit I thing to myself: "Shouldn't I be using `package_info` instead of `package`?

Sorry for wasting your time. Maybe you can somehow add warnings for dummies like me.

**Comment by memsharded** [comment]
No, no dummies here :) :)

I am trying to reduce the definition scope of ``cpp_info`` to the ``package_info()`` method in this PR: https://github.com/conan-io/conan/pull/2188

Thanks very much for the feedback!

**Comment by memsharded** [comment]
Hi Rick,

It seems it went good :) It has been merged to develop, it will be released in conan 1.0. Then, attempt to use it in other method than ``package_info()`` will be an error. Thanks again!

**Comment by Hedanito** [comment]
Yeah that should prevent it from happening again. Thank you very much!

**Comment by memsharded** [comment]
Done, and released in 1.0.0:  http://docs.conan.io/en/latest/conan1.0.html 

Please have a look, try it, and report as soon as possible, so things can be fixed before the final release. Thanks very much!
